### PR TITLE
8294368: Java incremental builds broken on Windows after JDK-8293116

### PR DESCRIPTION
**Pull request by JornVernee**
This patch fixes incremental builds on Windows.

There are 2 parts to this:
1. the build system needs to run the paths in the modified file list through fixpath. I've added a `convert` mode to `fixpath.sh` for that. There's an extra target for generating the file with fixed paths. On non-windows platforms this is just a simple `cp` of the file.
2. the dependency plugin of `javac` was using string-based path comparison. But, the paths fed by the build system and the paths used internally by javac could be in slightly different formats, meaning that files were not detected properly as changed. I switched to `Path`-based comparison instead and that fixes the issue.

Testing:
tested this manually by doing the following:
1. `make clean`
2. `make images`
3. put garbage in one of the files in `java.base`
4. `make images` (incremental)
5. verify that the build reported an error
6. verify the contents of `<build>\jdk\modules\java.base\_the.java.base_batch.modfiles.fixed`
7. revert the changes of 3, and do the same for another file
8. `make images` (incremental)
9. verify that the build reported an error
10. verify the contents of `<build>\jdk\modules\java.base\_the.java.base_batch.modfiles.fixed`
11. remove garbage from file modified by 9 again
12. `make images` (incremental)
13. verify that build succeeds as in 2

I've tested the build on Windows and Linux (WSL) using the above steps.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294368](https://bugs.openjdk.org/browse/JDK-8294368): Java incremental builds broken on Windows after JDK-8293116

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - Committer)
 * [Jan Lahoda](https://openjdk.org/census#jlahoda) (@lahodaj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10560/head:pull/10560` \
`$ git checkout pull/10560`

Update a local copy of the PR: \
`$ git checkout pull/10560` \
`$ git pull https://git.openjdk.org/jdk pull/10560/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10560`

View PR using the GUI difftool: \
`$ git pr show -t 10560`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10560.diff">https://git.openjdk.org/jdk/pull/10560.diff</a>

</details>

### CODE REVIEW

**Review by erikj79 — COMMENTED** [review]
This looks like a good fix. I would probably have done it slightly differently to avoid the redundant copy when it's not needed, but I think this is good enough. Thanks for fixing it!

File: make/scripts/fixpath.sh [hunk 1]
```diff
@@ -352,6 +352,20 @@ function convert_path() {
   fi
 }
 
+# Treat $1 as name of a file containing paths. Convert those paths to Windows style,
+# in the file specified by $2
+function convert_file() {
+  infile="$1"
+  outfile="$2"
+  if [[ -e $outfile ]] ; then
+	rm $outfile
```
[the remark this answers is not in the document]
Comment (JornVernee) [reply make/scripts/fixpath.sh]: Looks like some tabs snuck in. Fixed now

**Comment by JornVernee** [comment]
Thanks for the review, I've address your comments.

WRT the redundant copy. I wanted to avoid the use site having to have some `if` that checks whether we're running on Windows to decide which file `$1_COMPILE_TARGET` should depend on. Copying allows always depending on the fixed file.

**Comment by erikj79** [comment]
> WRT the redundant copy. I wanted to avoid the use site having to have some `if` that checks whether we're running on Windows to decide which file `$1_COMPILE_TARGET` should depend on. Copying allows always depending on the fixed file.

You are indeed correct and having a single code path is also a desirable attribute. Had this been in a more performance sensitive location, I would have still preferred to avoid the copy, but I think once per Java compilation unit is rare enough to not be an issue, especially since it's just unnecessary on non-windows.

**Review by erikj79 — APPROVED** [review]

**Review by djelinski — APPROVED** [review]
Also verified that:
- changes to public interface trigger full recompilation
- changes that do not affect public interface cause minimal recompilation.

This was on Cygwin. LGTM!

**Review by lahodaj — APPROVED** [review]
Thanks a lot for fixing this, and sorry for trouble!

**Pull request closed without merging**